tests/e2e/e2e_precompile_genesis_test.go

```diff
@@ -0,0 +1,171 @@
+//     sequence of 1, and no public key
+//   - An enabled precompile has code equal to 0x01
+//   - An enabled precompile has a nonce of 1
+//
+// This is important to ensure the genesis setup for precompiles is correct.
+func (suite *IntegrationTestSuite) TestPrecompileGenesis() {
+	type fixture struct {
+		address            string
+		expectIsEnabled    bool
+		expectIsEthAccount bool
+		expectCode         []byte
+		expectNonce        uint64
+	}
+
+	// enabled represnets the expected state for a registered precompile
+	// that is enabled at genesis
+	enabled := func(address string) func() fixture {
+		return func() fixture {
+			return fixture{
+				address:            address,
+				expectIsEnabled:    true,
+				expectIsEthAccount: true,
+				expectCode:         []byte{0x01},
+				expectNonce:        uint64(1),
+			}
+		}
+	}
+
+	// disabled represents the expected state for a registered precompile
+	// that is not enabled at genesis
+	disabled := func(address string) func() fixture {
+		return func() fixture {
+			return fixture{
+				address:            address,
+				expectIsEnabled:    false,
+				expectIsEthAccount: false,
+				expectCode:         []byte{},
+				expectNonce:        uint64(0),
+			}
+		}
+	}
+
+	testCases := []struct {
+		name       string
+		genFixture func() fixture
+	}{
+		{
+			name:       "noop contract address is enabled and initialized",
+			genFixture: enabled(registry.NoopContractAddress),
+		},
+		{
+			name:       "noop contract address second address is disabled and not initialized",
+			genFixture: disabled(registry.NoopContractAddress2),
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			tf := tc.genFixture()
+
+			//
+			// Addresses
+			//
+			evmAddress := common.HexToAddress(tf.address)
+			sdkAddress := sdk.AccAddress(evmAddress.Bytes())
+
+			//
+			// Heights
+			//
+			// We ensure all queries happen at block 1 after genesis
+			// and help ensure determisitc behavior
+			genesisHeight := big.NewInt(1)
+			grpcGenesisContext := suite.Kava.Grpc.CtxAtHeight(genesisHeight.Int64())
+
+			//
+			// Queries
+			//
+			evmParamsResp, err := suite.Kava.Grpc.Query.Evm.Params(grpcGenesisContext, &evmtypes.QueryParamsRequest{})
+			suite.Require().NoError(err)
+
+			// accountErr is checked during assertions if we are expecting an account due to
+			// the service returning a not found error for precompiles that are not enabled
+			accountResponse, accountErr := suite.Kava.Grpc.Query.Auth.Account(
+				grpcGenesisContext, &authtypes.QueryAccountRequest{Address: sdkAddress.String()})
+			var account authtypes.AccountI
+			if accountErr == nil {
+				err = suite.Kava.EncodingConfig.Marshaler.UnpackAny(accountResponse.Account, &account)
+				suite.Require().NoError(err)
+			}
+
+			// We ensure both the evm json rpc and x/evm grpc repeat the correct and same code value
+			grpcCodeResponse, err := suite.Kava.Grpc.Query.Evm.Code(grpcGenesisContext,
```

Comment:
```
// The GRPC endpoint returns []byte(nil) when the code is not in state, which is different from
// the rpc endpoint that returns []byte{}.
grpcExpectedCode := tf.expectCode
if len(grpcExpectedCode) == 0 {
    grpcExpectedCode = []byte(nil)
}
suite.Equalf(grpcExpectedCode, grpcCodeResponse.Code, "expected code for address %s to match expected", tf.address)
```

little easier to read then having to jump down to see where err is being handled.